internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -126,17 +127,21 @@ func RollbackWithOpts(ctx context.Context, log *logger.Logger, c client.Client,
 		return nil
 	}
 
+	if prevVersionedHome == "" {
+		prevVersionedHome = filepath.Join("data", fmt.Sprintf("%s-%s", AgentName, prevHash))
+	}
```

Comment:
Trying to understand: why is this needed? Could this could be set earlier in this function as part of the else clause on line 85? Also, instead of hardcoding "data", probably better to use `paths.DataFrom(topDirPath)`?

Reply:
Fixed in 52ab103953